### PR TITLE
[CIS-997] Fix over fetching previous messages

### PR DESCRIPTION
**Pull request by nuno-vieira**
## Description of the pull request
When getting the previous messages, we would make unnecessary requests. Now we only load the previous messages if there's more to fetch.

The only thing left to do is add test cases. I will do it once the implementation approach is approved.

### CODE REVIEW

**Review by b-onc — COMMENTED** [review]
I think this approach is good 👍

File: Sources/StreamChat/Workers/ChannelUpdater.swift [hunk 1]
```diff
@@ -17,19 +17,27 @@ class ChannelUpdater<ExtraData: ExtraDataTypes>: Worker {
...
                 let payload = try result.get()
                 channelCreatedCallback?(payload.channel.cid)
-                self.database.write { (session) in
+                self.database.write { session in
+                    try session.saveChannel(payload: payload)
+                    completion?(.success(payload))
+                }
+                self.database.write { session in
                     try session.saveChannel(payload: payload)
                 } completion: { error in
```
Comment (evsaev) [inline Sources/StreamChat/Workers/ChannelUpdater.swift:32]: why do we save channel twice? 🤔

File: Sources/StreamChat/Workers/ChannelUpdater.swift [hunk 2]
```diff
@@ -17,19 +17,27 @@ class ChannelUpdater<ExtraData: ExtraDataTypes>: Worker {
...
         apiClient.request(endpoint: .channel(query: channelQuery)) { (result) in
             do {
                 let payload = try result.get()
                 channelCreatedCallback?(payload.channel.cid)
-                self.database.write { (session) in
+                self.database.write { session in
+                    try session.saveChannel(payload: payload)
+                    completion?(.success(payload))
+                }
+                self.database.write { session in
```
Comment (evsaev) [inline Sources/StreamChat/Workers/ChannelUpdater.swift:30]: why do we save channel twice? 🤔

Comment (nuno-vieira) [reply Sources/StreamChat/Workers/ChannelUpdater.swift:30]: This seems wrong, nice catch 👍 Will fix it

Comment (nuno-vieira) [reply Sources/StreamChat/Workers/ChannelUpdater.swift]: @evsaev Done ✅

**Review by evsaev — APPROVED** [review]
Looks good 👍

**Pull request merged**